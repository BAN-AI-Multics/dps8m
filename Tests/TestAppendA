de dsbr.u 1
de dsbr.addr 020000
de ppr.ic 0
de ppr.prr 0
de ppr.psr 0
de ppr.p 1
dpsinit segment table
dpsinit SDWAM
load Play/Tests/PlayA.oct segment 0 address  0
load Play/Tests/UtilsA.oct segment 1 address  010000
set debug stderr
;set cpu debug=trace;appending;regdumpaqi;regdumpppr;addrmod
;set cpu debug=trace
dpsdump segment table
g 0

