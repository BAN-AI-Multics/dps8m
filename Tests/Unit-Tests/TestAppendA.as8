"
" code to test the appending unit ....
"
        go      Main
        name    Play

        segref  Utils, A1, V1
        
Main:
        tsp2    pr1|A1
        

        dis
    
        end
        