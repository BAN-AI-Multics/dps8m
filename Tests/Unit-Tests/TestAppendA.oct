!SIZE 000006
!LINKAGE 000002 000002
!SEGNAME Play
!SEGREF Utils A1 000002
!SEGREF Utils V1 000004
!GO 000000
000000 xxxx 100002272100         tsp2    pr1|A1
000001 xxxx 000000616000         dis
000002 xxxx 000000000046 link Utils|A1 
000003 xxxx 000000000000
000004 xxxx 000000000046 link Utils|V1 
000005 xxxx 000000000000
