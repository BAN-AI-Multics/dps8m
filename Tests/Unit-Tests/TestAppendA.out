
dps-8/m simulator V4.0-0 Beta        git commit id: f386776a
zero-initialized segments 0 .. 7
zero-initialized SDWAM
segment created for 'Play'
segdef created for segment Play, symbol 'Main', addr:000003
segref created for segment 'UtilsA' symbol:A0, addr:000226
segref created for segment 'UtilsA' symbol:A1, addr:000230
segref created for segment 'UtilsA' symbol:rd_clockA, addr:000232
segref created for segment 'UtilsA' symbol:ioa, addr:000234
Writing SDW to address 00020000 (DSBR.ADDR+2*0 offset) 
154 (000232) words loaded into segment 0(0) at address 000000
segment created for 'UtilsA'
segdef created for segment UtilsA, symbol 'A0', addr:000005
segdef created for segment UtilsA, symbol 'A1', addr:000050
segdef created for segment UtilsA, symbol 'rd_clockA', addr:000207
segdef created for segment UtilsA, symbol 'ioa', addr:000270
Writing SDW to address 00020002 (DSBR.ADDR+2*1 offset) 
851 (001523) words loaded into segment 1(1) at address 010000
Debug output to "STDERR"
*** Descriptor Segment Base Register (DSBR) ***
DSBR: ADDR=020000 BND=00000 U=1 STACK=0000
*** Descriptor Segment Table ***
Seg 0 - ADDR=000000 R1=0 R2=0 R3=0 F=1 FC=2 BOUND=12 R=1 E=1 W=1 P=1 U=1 G=1 C=1 EB=236
Seg 1 - ADDR=010000 R1=0 R2=0 R3=0 F=1 FC=2 BOUND=103 R=1 E=1 W=1 P=1 U=1 G=1 C=1 EB=2042
Seg 2 - *** Uninitialized ***
Seg 3 - *** Uninitialized ***
Seg 4 - *** Uninitialized ***
Seg 5 - *** Uninitialized ***
Seg 6 - *** Uninitialized ***
Seg 7 - *** Uninitialized ***
Date : ' 6/09 20:17 '
froms0=123
Sun , 9 Jun  2013 20:17:20 Pst 
Date 0: ' 6/09 20:17 '
This is A1... 3457623 -42 ASDF 000116 000045 | 000000 000043 000017 000000
desc1:1: 000116 000045, PR2: saved: 000000 000017, modified:000000 000116
1: If this works, I'll be surprised...
2: If this works, I'll be surprised...
This is the new ioa! -1234567890 123456 654321 7654321 12 345670 QwEr ABC123
4-bit decimal '0123456789+-'...
9-bit ASCII descriptor 'This is a test of the new inter-segment ioa!'
6-bit GEBCD descriptor 'THIS IS A TEST OF THE NEW INTER-SEGMENT IOA!'
'This is a test of the new inter-segment ioa!'
'THIS IS A TEST OF THE NEW INTER-SEGMENT IOA!'
'0123456789+-'
Well, I'll be ....

cpuCycles = 8675
TestAppendA-15> g 0

DIS instruction, IC: Play|114 (00000114) (000000616000)
Goodbye
Debug output disabled
